frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -846,10 +846,48 @@ export function ContainersSection(props: { resource: KubeObjectInterface | null
     return statuses;
   }
 
+  function getEphemeralContainerStatuses() {
+    if (!resource || resource.kind !== 'Pod') {
+      return {};
+    }
+
+    const ephemStatuses: {
+      [key: string]: ContainerInfoProps['status'];
+    } = {};
+
+    ((resource as KubePod).status.ephemeralContainerStatuses || []).forEach(
+      ephemeralContainerStatus => {
+        const { name, ...status } = ephemeralContainerStatus;
+        ephemStatuses[name] = { ...status };
+      }
+    );
+
+    return ephemStatuses;
+  }
+
+  function getInitContainerStatuses() {
+    if (!resource || resource.kind !== 'Pod') {
+      return {};
+    }
+
+    const initStatuses: {
+      [key: string]: ContainerInfoProps['status'];
+    } = {};
+
+    ((resource as KubePod).status.initContainerStatuses || []).forEach(initContainerStatus => {
+      const { name, ...status } = initContainerStatus;
+      initStatuses[name] = { ...status };
+    });
+
+    return initStatuses;
+  }
```

Comment:
Since this is repeated code from the getContainer function, we should abstract this into a helper, private function.